core/chains/evm/logpoller/orm.go

```diff
@@ -56,7 +56,7 @@ func (o *ORM) SelectBlockByNumber(n int64, qopts ...pg.QOpt) (*LogPollerBlock, e
 func (o *ORM) SelectLatestBlock(qopts ...pg.QOpt) (*LogPollerBlock, error) {
 	q := o.q.WithOpts(qopts...)
 	var b LogPollerBlock
-	if err := q.Get(&b, `SELECT * FROM log_poller_blocks WHERE block_number = (SELECT MAX(block_number) FROM log_poller_blocks) AND evm_chain_id = $1`, utils.NewBig(o.chainID)); err != nil {
+	if err := q.Get(&b, `SELECT * FROM log_poller_blocks WHERE evm_chain_id = $1 ORDER BY block_number LIMIT 1`, utils.NewBig(o.chainID)); err != nil {
```

Comment:
could you add `DESC` to specify the ordering? otherwise it's undefined behavior as far as I can tell https://stackoverflow.com/questions/8746519/sql-what-is-the-default-order-by-of-queries

Reply:
its not undefined (that looks like a mysql thread - see https://www.postgresql.org/docs/current/queries-order.html), but it does need to be desc not the default (asc) here, updated